### PR TITLE
[UBS user] Fix controller

### LINKED ISSUE #6059: [UBS user] Fix controller
End Point https://greencity-ubs.greencity.social/ubs/userProfile/user/update request a list of districts to update user.
It is unnecessary data for that.

### PR DESCRIPTION
**Pull request by ospodaryk**
# GreenCityUBS PR
https://github.com/ita-social-projects/GreenCity/issues/6059

## Summary Of Changes :fire:
Changed structure of AddressDto(Removed list of districts from that)

## Added
New Dto which contains List Of Districts and AddressDto

## Changed
All tests that had errors during my changes
Also changes Dtos where we have to get kist of districts(AddressDto to AddressWithDistrcitsDto)

## Deleted
Deleted from AddressDto list of districts because it is not always necessary 

# PR Checklist Forms
- [x] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

**Review by 2or5 — APPROVED** [review]

**Review by sergoliarnik — APPROVED** [review]

File: core/src/test/java/greencity/ModelUtils.java [hunk 1]
```diff
@@ -69,11 +71,7 @@
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.Month;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
```
Comment (VadymMakitra) [inline core/src/test/java/greencity/ModelUtils.java]: Please, don't use wildcard import

File: service-api/src/main/java/greencity/dto/address/AddressWithDistrictsDto.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import greencity.dto.location.api.DistrictDto;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import java.io.Serializable;
+import java.util.List;
+
+@Builder
+@Data
```
Comment (VadymMakitra) [inline service-api/src/main/java/greencity/dto/address/AddressWithDistrictsDto.java:12]: Is it required to have a builder and data?

File: service/src/test/java/greencity/mapping/location/AddressToAddressWithDistrictsDtoMapperTest.java [hunk 3]
```diff
@@ -31,15 +21,15 @@ class AddressToAddressDtoMapperTest {
...
     @Test
     void convert() {
         MockitoAnnotations.initMocks(this);
-        AddressDto expected = ModelUtils.getAddressDto(1L);
+        AddressWithDistrictsDto expected = ModelUtils.getAddressWithDistrictsDto(1L);
         when(locationApiService.getAllDistrictsInCityByNames(anyString(), anyString()))
             .thenReturn(ModelUtils.getLocationApiDtoList());
-        AddressDto actual = addressToAddressDtoMapper.convert(ModelUtils.getAddress(1L));
+        AddressWithDistrictsDto actual = addressToAddressWithDistrictsDtoMapper.convert(ModelUtils.getAddress(1L));
         assertEquals(expected.getAddressRegionDistrictList().size(), actual.getAddressRegionDistrictList().size());
```
Comment (VadymMakitra) [inline service/src/test/java/greencity/mapping/location/AddressToAddressWithDistrictsDtoMapperTest.java:33]: Please, also add assert to check content inside the list.

File: service/src/test/java/greencity/service/ubs/UBSClientServiceImplTest.java [hunk 4]
```diff
@@ -1669,15 +1670,15 @@ void testFindAllAddressesForCurrentOrder() {
...
 
         OrderWithAddressesResponseDto expected = new OrderWithAddressesResponseDto(testAddressesDto);
 
         List<Address> addresses = getTestAddresses(user);
 
         when(addressRepository.findAllNonDeletedAddressesByUserId(user.getId())).thenReturn(addresses);
-        when(modelMapper.map(addresses.get(0), AddressDto.class)).thenReturn(testAddressesDto.get(0));
-        when(modelMapper.map(addresses.get(1), AddressDto.class)).thenReturn(testAddressesDto.get(1));
+        when(modelMapper.map(addresses.get(0), AddressWithDistrictsDto.class)).thenReturn(testAddressesDto.get(0));
+        when(modelMapper.map(addresses.get(1), AddressWithDistrictsDto.class)).thenReturn(testAddressesDto.get(1));
```
Comment (VadymMakitra) [inline service/src/test/java/greencity/service/ubs/UBSClientServiceImplTest.java:1671]:
```suggestion
for(int i =0; i < addresses.size(); i ++) {
        when(modelMapper.map(addresses.get(i), AddressWithDistrictsDto.class)).thenReturn(testAddressesDto.get(i));
        }
```

File: service/src/test/java/greencity/service/ubs/UBSClientServiceImplTest.java [hunk 5]
```diff
@@ -1702,6 +1703,47 @@ private List<Address> getTestAddresses(User user) {
...
+            .coordinates(Coordinates.builder().latitude(2.3).longitude(5.6).build())
+            .district("Залізничний")
+            .districtEn("Zaliznuchnuy")
+            .regionEn("Region")
+            .region("Регіон")
+            .cityEn("Lviv")
+            .city("Львів")
+            .actual(false).build())
+            .addressRegionDistrictList(new ArrayList<>())
+
```
Comment (VadymMakitra) [inline service/src/test/java/greencity/service/ubs/UBSClientServiceImplTest.java]: please remove blank line

File: service/src/test/java/greencity/service/ubs/UBSClientServiceImplTest.java [hunk 6]
```diff
@@ -1702,6 +1703,47 @@ private List<Address> getTestAddresses(User user) {
...
+            .streetEn("Shevchenka")
+            .coordinates(Coordinates.builder().latitude(3.3).longitude(6.6).build())
+            .district("Залізничний")
+            .districtEn("Zaliznuchnuy")
+            .regionEn("Region")
+            .region("Регіон")
+            .city("Львів")
+            .cityEn("Lviv").actual(false).build())
+            .addressRegionDistrictList(new ArrayList<>())
+
```
Comment (VadymMakitra) [inline service/src/test/java/greencity/service/ubs/UBSClientServiceImplTest.java]: please remove blank line

File: service/src/main/java/greencity/mapping/user/AddressToAddressWithDistrictsDtoMapper.java [hunk 7]
```diff
@@ -51,8 +52,9 @@ public AddressDto convert(Address address) {
                 .latitude(address.getCoordinates().getLatitude())
                 .longitude(address.getCoordinates().getLongitude())
                 .build())
+            .actual(address.getActual()).build())
             .addressRegionDistrictList(getAllDistricts((address.getRegion()), address.getCity()))
-            .actual(address.getActual())
+
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/mapping/user/AddressToAddressWithDistrictsDtoMapper.java]: please remove blank line

File: core/src/test/java/greencity/ModelUtils.java [hunk 8]
```diff
@@ -160,9 +157,79 @@ public static List<AddressDto> addressDto() {
...
+            .streetEn("Shevchenka")
+            .coordinates(Coordinates.builder().latitude(3.3).longitude(6.6).build())
+            .district("Залізничний")
+            .districtEn("Zaliznuchnuy")
+            .regionEn("Region")
+            .region("Регіон")
+            .city("Львів")
+            .cityEn("Lviv").actual(false).build())
+            .addressRegionDistrictList(new ArrayList<>())
+
```
Comment (VadymMakitra) [inline core/src/test/java/greencity/ModelUtils.java]: please remove blank line

File: service-api/src/main/java/greencity/dto/address/AddressWithDistrictsDto.java [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+@NoArgsConstructor
+@AllArgsConstructor
+public class AddressWithDistrictsDto implements Serializable {
+    private AddressDto addressDto;
```
Comment (VadymMakitra) [inline service-api/src/main/java/greencity/dto/address/AddressWithDistrictsDto.java:16]: please add serialVersionUID

**Review by juseti — APPROVED** [review]

**Pull request merged**